services/blockchain-connector/shared/sdk/genesisBlock.js

```diff
@@ -83,9 +84,68 @@ const getGenesisConfig = async () => {
 	}
 };
 
+const getGenesisAssets = async (params = {}) => {
+	const rawGenesisBlock = await getGenesisBlock(true);
+	const genesisBlock = await formatBlock(rawGenesisBlock);
+
+	const assetModule = genesisBlock.assets.find(
+		genesisAsset => genesisAsset.module === params.module,
+	);
+
+	if (params.subStore) {
+		const assetModuleSubStoreKey = Object.keys(assetModule.data).find(
+			subStoreKey => subStoreKey === params.subStore,
+		);
+
+		return assetModuleSubStoreKey
+			? [{
+				...assetModule,
+				data: {
+					[assetModuleSubStoreKey]: assetModule.data[assetModuleSubStoreKey],
+				},
+			}]
+			: [];
+	}
+
+	if (params.module) {
+		return assetModule ? [assetModule] : [];
+	}
+
+	return genesisBlock.assets;
+};
+
+/* Returns a nested object of following structure filtered by module and subStore if present
+{
+	moduleName1: {
+		module1.data.key1 (subStore): module1.data.key1.length,
+	}
+	moduleName2: {
+		module2.data.key2 (subStore): module2.data.key2.length,
+	}
+}
+*/
+const getGenesisAssetsLength = async (params) => {
+	const genesisAssets = await getGenesisAssets(params);
```

Comment:
This would not return the desired response when no params are passed. Please re-verify. Also, the corresponding tests are missing.

Reply:
Seems to respect the response structure. This the response when no empty params is passed. Will add test coverage.

```
{
    "interoperability":{
        "outboxRootSubstore":0,
        "chainDataSubstore":0,
        "channelDataSubstore":0,
        "chainValidatorsSubstore":0,
        "ownChainDataSubstore":1,
        "terminatedStateSubstore":0,
        "terminatedOutboxSubstore":0,
        "registeredNamesSubstore":1
    },
    "legacy":{
        "accounts":30
    },
    "pos":{
        "validators":103,
        "stakers":103
    },
    "token":{
        "userSubstore":104,
        "supplySubstore":1,
        "escrowSubstore":0,
        "supportedTokensSubstore":0
    }
}
```